[12.0][ADD] product_category_active: new module

ernestotejeda opened the pull request:
This module allows you to archive product categories

Cc @Tecnativa TT23505

Commit:
[ADD] product_category_archive: new module

chienandalu commented on product_category_archive/README.rst:
```diff
@@ -0,0 +1,87 @@
+#. Go to *Inventory > Configuration > Products > Product categories*.
+#. Select a product category or create a new one.
+#. Click on the smart button Archive (this action will set active
+   field to False).
```
Note that if there are products in the category, it can't be archived.

chienandalu reviewed (approved):
Code and runbot :+1: Just a little remark:

legalsylvain reviewed (commented):
Hi all. 

A general question regarding the name of the module : Many options are present for the time being in OCA for the name, ``_archive``, ``_active``, ``_inactive`` suffixes : 

- ``account_type_inactive``
- ``res_company_active``, ``partner_bank_active``, ``project_task_type_active`` (and coming soon ``account_template_active`` [here](https://github.com/OCA/account-financial-tools/pull/978/))
- ``sale_order_archive`` ;  ``purchase_order_archive``

Generally, it seems that the tecnical name of a module, is related to the technical concept, not the business name (ie = modules contains ``partner``, not ``contact`` for exemple)
So I'm in favor to switch for ``product_category_active``. What do you think ?

kind regards.

pedrobaeza commented:
OK, @ernestotejeda please go to `product_category_active`.

pedrobaeza commented:
Edited last comment as I have mixed the name.

legalsylvain reviewed (commented):
Code review. LGTM.
One improvement I see : could you
- add the active field in the end of the tree view with decoration-muted if inactive and toggle widget
- add active_test = False in the related action

(You can copy paste from the module ``account_template_active`` i mentioned)

Thanks !

TT 23505 : is the number of OCA module Tecnativa developed ?

legalsylvain reviewed (approved):
thanks for the changes and thanks for this contribution !

pedrobaeza commented on product_category_active/models/product.py:
```diff
@@ -0,0 +1,28 @@
+        if self.env['product.template'].search([
+            '|',
+            ('categ_id', 'in', to_archive.ids),
+            ('categ_id', 'child_of', to_archive.ids),
```
No need of this double condition. `child_of` operator already includes it. If you put `child_of, [3, 5]`, it means that it will return `3, 5`, or any of their childrens.
```suggestion
            ('categ_id', 'child_of', to_archive.ids),
```

pedrobaeza commented on product_category_active/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,2 @@
+This module allows you to archive product categories
```
```suggestion
This module allows you to archive product categories.
```

pedrobaeza commented on product_category_active/readme/USAGE.rst:
```diff
@@ -0,0 +1,9 @@
+#. Click on the smart button Archive (this action will set active
+   field to False).
+
+Note: If there are products in a category, it can't be archived.
```
```suggestion
Note: If there are products in the category to be archived or any of its children categories,
en error will be raised and it won't be archived.

Note 2: Children category are not archived/unarchived when the parent category does.
```

pedrobaeza committed:
[FIX] product_category_archive: Constraint search refinement

The problem why child_of was not working directly in the search is because there are
involved inactive records that are excluded by default by the ORM.

We need to disable that exlusion for a proper working, and it's needed because the
other domain is not taking into account possible inactive categories in the middle
of the hierarchy.

pedrobaeza commented:
Push a fix on the constraint domain with `active_test`, so everything OK by my part:

/ocabot merge nobump
